### PR TITLE
Fix new matches

### PR DESCRIPTION
**Pull request by gesundkrank**

### CODE REVIEW

File: frontend/src/controllers/tournamentController.ts [hunk 1]
```diff
@@ -56,24 +56,37 @@ export class TournamentController {
...
   }
 
-  checkForRunningMatch() {
-    return this.pull().then(() => this.getRunningMatch());
+  checkAndUpdateTournamentState(): Promise<State> {
+    if (this.tournament.state == State.FINISHED) {
+      return Promise.resolve(State.FINISHED);
+    }
+    const wins = this.countWins(this.tournament.matches);
+    const maxWins = this.tournament.bestOfN / 2 + 1;
```
Comment (hvoecking) [inline frontend/src/controllers/tournamentController.ts:64]: Why are we still dealing with `bestOfN` if `N` is always 3?

File: frontend/src/controllers/tournamentController.ts [hunk 2]
```diff
@@ -56,24 +56,37 @@ export class TournamentController {
...
+
+    if (Math.max(wins['teamA'], wins['teamB']) >= maxWins) {
+      this.finishTournament().then(() => this.tournament.state);
```
Comment (hvoecking) [inline frontend/src/controllers/tournamentController.ts:67]: Why are you having a `then` here? It looks like the return value is not used anyways

File: frontend/src/controllers/tournamentController.ts [hunk 3]
```diff
@@ -56,24 +56,37 @@ export class TournamentController {
...
         }
-        const running = this.findRunning();
-        if (!running) {
-          if (tournament.state == State.RUNNING) {
-            return this.newMatch().then(tournament => this.getRunningMatch())
+
+        return this.checkAndUpdateTournamentState().then(state => {
+          const running = this.findRunning();
+          if (!running) {
+            if (state == State.RUNNING) {
```
Comment (hvoecking) [inline frontend/src/controllers/tournamentController.ts:83]: Do you really really really want to use `==`??

File: frontend/src/controllers/tournamentController.ts [hunk 4]
```diff
@@ -56,24 +56,37 @@ export class TournamentController {
...
+              return this.newMatch().then(tournament => this.getRunningMatch())
+            }
+            return;
           }
-          return;
-        }
-        return running;
+          return running;
```
Comment (hvoecking) [inline frontend/src/controllers/tournamentController.ts:88]: How about return ealry?
```
if (running) {
  return running;
}
if (state === State.RUNNING) {
  return this.newMatch().then(tournament => this.getRunningMatch());
}

File: frontend/src/controllers/tournamentController.ts (hunk 3 above)
Comment (gesundkrank) [reply frontend/src/controllers/tournamentController.ts:83]: Of course

File: frontend/src/controllers/tournamentController.ts (hunk 1 above)
Comment (gesundkrank) [reply frontend/src/controllers/tournamentController.ts:64]: It should still be configurable and not hard coded in a thousand places.

**Pull request merged**